### PR TITLE
Add pause state

### LINKED ISSUE #11: Pause / Unpause Functionality
- [ ] The contract should have a pause state that defaults to false
- [x] The creator or the admin can set the pause state, and should pass true or false when setting (i.e. not a toggle)
- [x] No other address should be able to call pause
- [x] When the contract is Paused/Unpaused a `ToggledPause` should be emitted with the old state, new state, and the address that called the method
- [ ] While paused the buy and sell methods should revert

### PR DESCRIPTION
**Pull request by garyghayrat**
closes #11

### CODE REVIEW

File: src/CreatorToken.sol [hunk 1]
```diff
@@ -4,13 +4,15 @@ pragma solidity 0.8.20;
...
+import {Pausable} from "openzeppelin/security/Pausable.sol";
 
-contract CreatorToken is ERC721 {
+contract CreatorToken is ERC721, Pausable {
   using SafeERC20 for IERC20;
 
   error CreatorToken__MaxPaymentExceeded(uint256 _price, uint256 _maxPayment);
   error CreatorToken__CallerIsNotCreator(address _creator, address _caller);
   error CreatorToken__CallerIsNotAdmin(address _admin, address _caller);
+  error CreatorToken__CallerIsNotCreatorOrAdmin(address _creator, address _admin, address _caller);
```
Comment (apbendi) [inline src/CreatorToken.sol:15]: Let's collapse these three errors into one `CreatorToken_Unauthorized(bytes32 reason, address caller)`, where reason is either "not admin", "not creator", or "not creator or admin".

**Review by apbendi — COMMENTED** [review]
hey @garyghayrat, some feedback. Sorry for the churn!

File: src/CreatorToken.sol [hunk 2]
```diff
@@ -4,13 +4,15 @@ pragma solidity 0.8.20;
 import {ERC721} from "openzeppelin/token/ERC721/ERC721.sol";
 import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
 import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {Pausable} from "openzeppelin/security/Pausable.sol";
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply src/CreatorToken.sol:7]: Sounds good, removed it here and now using our own solution: https://github.com/showtime-xyz/creator-tokens/pull/12/commits/310ca54770703a02b6bc34ae08dc1bab2b224243

File: src/CreatorToken.sol (hunk 1 above)
Comment (garyghayrat) [reply src/CreatorToken.sol]: updated in the same commit as above: https://github.com/showtime-xyz/creator-tokens/commit/310ca54770703a02b6bc34ae08dc1bab2b224243

**Pull request merged**